### PR TITLE
Detekt

### PR DESCRIPTION
**Pull request by quietbits**

### CODE REVIEW

File: config/detekt/detekt.yml [hunk 1]
```diff
@@ -0,0 +1,741 @@
+build:
```
Comment (Ifropc) [inline config/detekt/detekt.yml:1]: Is it a default config? Can you please delete it and only keep changes that are different from a default config?

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Anchor.kt [hunk 2]
```diff
@@ -167,7 +167,8 @@ internal constructor(
     val request = OkHttpUtils.buildStringGetRequest(endpointUrl, authToken)
 
     log.debug {
-      "Anchor account's all transactions request: assetCode = $assetCode, authToken = ${authToken.take(8)}"
+      "Anchor account's all transactions request: assetCode = $assetCode, authToken = ${authToken
+        .take(STRING_TRIM_LENGTH)}"
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Anchor.kt:171]: I think we can remove "magic numbers" rule. I find it too strict, a lot of the time it's obvious from the context where it comes from

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Anchor.kt [hunk 3]
```diff
@@ -167,7 +167,8 @@ internal constructor(
     val request = OkHttpUtils.buildStringGetRequest(endpointUrl, authToken)
 
     log.debug {
-      "Anchor account's all transactions request: assetCode = $assetCode, authToken = ${authToken.take(8)}"
+      "Anchor account's all transactions request: assetCode = $assetCode, authToken = ${authToken
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Anchor.kt:170]: [nits] can you add a (value) type for AuthToken and an extension function for .take(8)? Something like `AuthToken.prettify()`

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/exception/AnchorExceptions.kt [hunk 4]
```diff
@@ -25,11 +25,12 @@ class IncorrectTransactionStatusException(
   expectedStatus: String
 ) :
   WalletException(
-    "Incorrect status of transaction ${transaction.id}. Expected status to be $expectedStatus and transaction has status ${transaction.status}"
+    "Incorrect status of transaction ${transaction.id}. Expected status to be $expectedStatus and" +
+            " transaction has status ${transaction.status}"
   )
 
-class AnchorAuthNotSupported() :
+class AnchorAuthNotSupported :
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/exception/AnchorExceptions.kt:32]: [nit] on this 2: can be objects

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Data.kt [hunk 5]
```diff
@@ -6,6 +6,7 @@ import org.stellar.sdk.Memo
 import org.stellar.walletsdk.ApplicationConfiguration
 import org.stellar.walletsdk.util.Util.isHex
 
+@Suppress("ConstructorParameterNaming")
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Data.kt:9]: I think you should just constructor parameters

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt [hunk 6]
```diff
@@ -1,9 +1,25 @@
...
 
-@Serializable data class ChallengeResponse(val transaction: String, val network_passphrase: String)
+@Suppress("ConstructorParameterNaming")
+@Serializable
+data class ChallengeResponse(val transaction: String, val network_passphrase: String)
 
-@Serializable internal data class AuthToken(val token: String)
+@Serializable
+@JvmInline
+value class AuthTokenValue(private val value: String) {
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt:13]: Rename to AuthToken?

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt [hunk 7]
```diff
@@ -1,9 +1,25 @@
...
+  fun prettify(): String {
+    return value.take(STRING_TRIM_LENGTH)
+  }
+
+  override fun toString(): String {
+    return value
+  }
+}
+
+@Serializable internal data class AuthToken(@Contextual val token: AuthTokenValue)
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt:23]: I'm not sure if you even need this class now

**Review by Ifropc — COMMENTED** [review]
Can you add git commit hook that checks detekt and spotless? So we don't commit bad code by mistake?

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt (hunk 7 above)
Comment (quietbits) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt:23]: Can we use `AuthToken` both for value and response for the auth which returns `token`?

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Wallet.kt [hunk 8]
```diff
@@ -20,6 +20,7 @@ class Wallet(
   internal val cfg = Config(stellarConfiguration, applicationConfiguration)
 
   init {
+    @Suppress("UseRequire")
```
[the remark this answers is not in the document]
Comment (quietbits) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Wallet.kt:23]: I tried to do that, but it broke things. I can give it another try.

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Data.kt (hunk 5 above)
Comment (quietbits) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Data.kt:9]: Can we use something like `@SerialName("min_amount") val minAmount` for all of them?

Comment (Ifropc) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Data.kt:9]: Yes

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Wallet.kt (hunk 8 above)
[the remark this answers is not in the document]
Comment (Ifropc) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Wallet.kt:23]: IIRC you just need to flip the logic in require 
i.e. if (condition) { throw } becomes require (!condition)

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt (hunk 7 above)
Comment (Ifropc) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt:23]: If it returns object, you may still need this data class wrapper (or you can make a custom deserializer) 
Probably easier to just leave a wrapper, but please rename AuthTokenValue -> AuthToken and AuthToken -> AuthTokenResponse

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Wallet.kt (hunk 8 above)
[the remark this answers is not in the document]
Comment (quietbits) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Wallet.kt]: I think I got it working. :crossed_fingers:

**Review by Ifropc — APPROVED** [review]

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/horizon/AccountService.kt [hunk 9]
```diff
@@ -25,6 +26,7 @@ internal constructor(
 ) {
   private val server: Server = cfg.stellar.server
   private val network: Network = cfg.stellar.network
+  @Suppress("UnusedPrivateMember")
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/horizon/AccountService.kt:29]: If it's not used you can just delete it

**Pull request merged**